lib/ansible/modules/cloud/podman/podman_container.py

```diff
@@ -0,0 +1,1281 @@
+            group_add=dict(type='str'),
+            healthcheck=dict(type='str'),
+            healthcheck_interval=dict(type='str'),
+            healthcheck_retries=dict(type='int'),
+            healthcheck_start_period=dict(type='str'),
+            healthcheck_timeout=dict(type='str'),
+            hostname=dict(type='str'),
+            http_proxy=dict(type='bool'),
+            image_volume=dict(type='str', choices=['bind', 'tmpfs', 'ignore']),
+            init=dict(type='str'),
+            init_path=dict(type='str'),
+            interactive=dict(type='bool'),
+            ip=dict(type='str'),
+            ipc=dict(type='str'),
+            kernel_memory=dict(type='str'),
+            label=dict(type='dict', default={}),
+            label_file=dict(type='str'),
+            log_opt=dict(type='str', aliases=['log_options']),
+            memory=dict(type='str'),
+            memory_reservation=dict(type='str'),
+            memory_swap=dict(type='str'),
+            memory_swappiness=dict(type='int'),
+            mount=dict(type='str'),
+            network=dict(type='str', aliases=['net']),
+            no_hosts=dict(type='bool'),
+            oom_kill_disable=dict(type='bool'),
+            oom_score_adj=dict(type='int'),
+            pid=dict(type='str'),
+            pids_limit=dict(type='str'),
+            pod=dict(type='str'),
+            privileged=dict(type='bool'),
+            publish=dict(type='list', elements='str', aliases=['ports', 'published', 'published_ports'], default=[]),
+            publish_all=dict(type='bool'),
+            read_only=dict(type='bool'),
+            read_only_tmpfs=dict(type='bool'),
+            restart_policy=dict(type='str'),
+            rm=dict(type='bool', aliases=['remove']),
+            rootfs=dict(type='bool'),
+            security_opt=dict(type='str'),
+            shm_size=dict(type='str'),
+            sig_proxy=dict(type='bool'),
+            stop_signal=dict(type='str'),
+            stop_timeout=dict(type='int'),
+            subuidname=dict(type='str'),
+            subgidname=dict(type='str'),
+            sysctl=dict(type='str'),
+            systemd=dict(type='bool'),
+            tmpfs=dict(type='str'),
+            tty=dict(type='bool'),
+            uidmap=dict(type='str'),
+            ulimit=dict(type='str'),
+            user=dict(type='str'),
+            userns=dict(type='str'),
+            uts=dict(type='str'),
+            volume=dict(type='list', elements='str', aliases=['volumes'], default=[]),
+            volumes_from=dict(type='list', elements='str', default=[]),
+            workdir=dict(type='str'),
+        ),
+        mutually_exclusive=(
+            ['no_hosts', 'etc_hosts'],
+
+        ),
+    )
+    # work on input vars
+    if module.params['state'] in ['started', 'present'] and not module.params['image']:
+        module.fail_json(msg="State '%s' required image to be configured!" % module.params['state'])
+
+    results = dict(
```

Comment:
1) There's no need in initializing this outside of the `PodmanContainer` class
2) Please use dict literals over the contructor style

Reply:
fixed